clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -381,13 +381,12 @@ public void testEpochUpdateOnChangedTopicIds() {
         metadata.updateWithCurrentRequestVersion(metadataResponse, false, 0L);
 
         // Start with a topic with no topic ID
-        metadataResponse = RequestTestUtils.metadataUpdateWith("dummy", 1, Collections.emptyMap(), Collections.singletonMap("topic-1", 1), _tp -> 10);
+        metadataResponse = RequestTestUtils.metadataUpdateWith("dummy", 1, Collections.emptyMap(), Collections.singletonMap("topic-1", 1), _tp -> 100);
         metadata.updateWithCurrentRequestVersion(metadataResponse, false, 1L);
-        assertEquals(Optional.of(10), metadata.lastSeenLeaderEpoch(tp));
+        assertEquals(Optional.of(100), metadata.lastSeenLeaderEpoch(tp));
 
-        // We should treat an added topic ID as though it is the same topic. Handle only when epoch increases.
-        // Don't update to an older one
-        metadataResponse = RequestTestUtils.metadataUpdateWithIds("dummy", 1, Collections.emptyMap(), Collections.singletonMap("topic-1", 1), _tp -> 1, topicIds);
+        // If the Older topic Id is null, we should go with the new TopicId as the leader Epoch
```

Comment:
It might be worthwhile having a separate case which goes through the sequence described in the jira. Basically this:
1. Receive metadata response with topicID A.
2. Receive metadata response with UNKNOWN_TOPIC error.
3. Receive metadata response with topicID B.

Reply:
added a new testcase with the suggested flow. Old test-case still needs to be changed as that case fails now, so modified the case as per the new changes.

Reply:
nit: `Older` -> `older`; `topic Id` -> `topic ID`; `TopicId` -> `topic ID`.